account_move_line_search_extension/res_partner.py

```diff
@@ -29,13 +29,7 @@ class res_partner(models.Model):
     def search(self, cr, uid, args,
                offset=0, limit=None, order=None, context=None, count=False):
         if context and 'account_move_line_search_extension' in context:
-            args.extend(
-                ['|',
-                 ('parent_id', '=', False),
-                 ('is_company', '=', True),
-                 '|',
-                 ('active', '=', False),
-                 ('active', '=', True)])
+            args.extend(['|', ('active', '=', False), ('active', '=', True)])
```

Comment:
Odoo's "not-so-perfect" res_partner implementation allows following operation:
- create a partner without "is_company" flag
- create an accounting entry (e.g. via invoice) for that partner
- afterwards you realise that this partner is actually a contact that belongs to a company hence you link the contact to an "is_company" partner record.

Now you have a (one of the many) confusing situation which is the consequence of Odoo's res_partner model introduced since V7. 
Your database has an accounting entry linked to partner that shouldn't have accounting entries.

The ambition of the 'account_move_line_search_extension' model is not to fix basic res_partner errors but to facilitate the search of accounting entries, hence we should also cover this situation.